### ISSUE TITLE
View score in popup and shade by score for GFF tracks

### ISSUE THREAD

**Issue by Sammi-Smith** [igvteam/igv.js#1466]
Our team's software outputs information about ROIs (regions of interest) in a GFF3 file, including various scores associated with each region.  

I am displaying this file using the `roi` property at the root level (to highlight the region across all tracks that are being displayed) as well as on its own `annotation` track (so the user can click on a region and get more info with the popup, including the various types of scores included as attributes in GFF column 9).

This is what the first few lines of my GFF file look like:

```
##gff-version 3
##sequence-region 1_pilon 1 4215054
1_pilon	feature	peak	1650596	1665689	0.004226614290639252	+	.	ROI Detected=1;engineered_read_rate=0.03225806451612903;peak_prominences=0.010518934081346423
1_pilon	feature	peak	3575146	3590145	0.004866576525554385	+	.	ROI Detected=1;engineered_read_rate=0.029411764705882353;peak_prominences=0.0030959752321981435
1_pilon	feature	peak	3444831	3459830	0.004917009909145034	+	.	ROI Detected=1;engineered_read_rate=0.022222222222222223;ki_cluster_density_e_value=0.14528782662300455;peak_prominences=0.0029914529914529912
1_pilon	feature	peak	3486328	3501327	0.004917009909145034	+	.	ROI Detected=1;engineered_read_rate=0.023255813953488372;ki_cluster_density_e_value=0.13851412941883356;peak_prominences=0.0019792182088075212
```
And this is how I am loading in the file as an ROI at the root level and as an annotation track. The variable "url_roi" gives the location of the above GFF file.  

```
  //specify browser config
  options = {
      reference: {            
                      id: sample_id,
                      name: 'Assembly',
                      fastaURL: fastaURL_preprocessing_genome_assembly,
                      indexURL: indexURL_preprocessing_genome_assembly, 
      },               
      tracks: [

                  {
                      name: "ROI",
                      url: url_roi,
                      type: 'annotation',
                      displayMode: "EXPANDED",
                      height: 50,
                      color: "rgba(150, 0, 0, 0.9)",
                      order: 0 //pin this track to the top
                  },

                  {//other tracks added here

                  },

              ],
      // Define global (across all tracks) "region of interest" set
      roi: [
                {
                    name: "ROI",
                    url: url_roi,
                    color: "rgba(150, 0, 0, 0.1)"
                }
            ]
  };

  browser = igv.createBrowser(div, options);
}
```
I have two related questions:

1. How can the score in column 6 be displayed on the popup for the annotation track?  As you can see in the below screenshot, I'm only seeing the Type (column 3), Source (column 2), Location (columns 1, 4 and 5), and the 3 attributes present in column 9, but the score in column 6 is not showing.
![image](https://user-images.githubusercontent.com/49876975/155015838-dec4fbaa-2bae-42a7-ac24-6bd44a9e3369.png)

2. How can the ROIs displayed at root level use the column 6 score values to shade the regions according to the score, if the GFF file includes a track line with the option `useScore=1`?  I see that #869 made this option available for BED files, but it doesn't seem to work for GFF files.  I'd also like to do this shading according to score in the annotation track, if possible.

**Comment by jrobinso** [comment]
There's a bug, and a feature request here. 

(1) The missing score in the popup is a bug, I will add it.

(2) The UCSC "useScore" attribute is a UCSC concept for UCSC formats, and works on a standardized value between 0 and 1000.   By UCSC conventions your scores are all essentially zero.   This could be generalized to work off some other scale, and work with non-UCSC formats such as GFF that have a defined "score" field.  I will leave this issue open as a request for GFF "score" shading

I think, but am not sure, when "score" shading is supported for GFF tracks it will automatically work for ROI as well.

I

**Comment by jrobinso** [comment]
BTW of all the gff's I've seen, probably thousands, this is the first to have a score other than "."

**Comment by Sammi-Smith** [comment]
Also regarding (2): Rather than using track lines and the `useScore` option, another way to implement score shading for GFF tracks might be to make use of the `colorBy` option for annotation tracks.

**Comment by Sammi-Smith** [comment]
Yes, I've actually been experimenting with that already this morning!  I was building off your [example in the Annotation Track Wiki](https://github.com/igvteam/igv.js/wiki/Annotation-Track#color-function-igvjs-version-2101) that uses a function for the `color` field.  Following that example, I was using `feature.getAttributeValue("score")` which worked as far as displaying the color as expected, even though it returned the below errors.  I get the same errors even if I use one of the actual attributes from column 9, like `feature.getAttributeValue("engineered_read_rate")`.
![image](https://user-images.githubusercontent.com/49876975/155197170-fb607bdf-f681-48e6-a4d6-16b44592e3ab.png)

When I changed it to `feature.score` as you just recommended, those errors disappear!  LIke I said, the syntax on that wiki page still returns the value as expected even though it throws the error, so not exactly sure what's going on with that.

This is the function I'm passing in to the `color` field for the annotation track.

```
                  color: (feature) => {
                    let score = feature.score;
                    if (score > .75) {
                        colorValue = "rgba(150, 0, 0, 1)"
                    } else if (score > .5) {
                            colorValue = "rgba(150, 0, 0, 0.75)"
                        } else if (score > .25) {
                                colorValue = "rgba(150, 0, 0, 0.5)"
                            } else {
                                colorValue = "rgba(150, 0, 0, 0.1)"
                            }
                    return colorValue;
                    },
```

Here's what it looks like so far!  You can see how most of the regions are shown in the most transparent shade of red, but a couple of them stand out as darker shades of red.  This is EXACTLY what we were hoping for - we want the user to be able to quickly glace to see which ROIs are most... well, interesting. :stuck_out_tongue_winking_eye: (where higher score = more interesting)
![image](https://user-images.githubusercontent.com/49876975/155199722-85f34562-51c3-467d-b929-d5a22c93c95e.png)

I tried passing in the same color function to the `color` field of the ROIs displayed at root level, but weirdly enough, it just caused all ROI regions across all tracks to be displayed as completely opaque black bars.  I'm guessing there is something about ROIs that doesn't support passing in a function to the `color` field?  For now, I'm just passing in `color: "rgba(0, 0, 0, 0.05)"` to display them all as the same shade of light grey so they don't obscure information in the other tracks.

**Comment by jrobinso** [comment]
I don't know what the ROI problem is, leave this open for that.

WRT score, its subtle and not documented but the first 8 columns are properties of the feature, so you access them directly, as in feature.score.   The column 9 key-values are held in an attributes dictionary, so you fetch an attribute by name (key).

**Comment by Sammi-Smith** [comment]
Hi Jim,

Circling back to this, I updated my code this week to use the latest version that you tagged v2.11.0, since I saw that incorporated the code for the bugfix. This is what I'm using now:

`<script type="text/javascript" src="https://cdn.jsdelivr.net/npm/igv@2.11.0/dist/igv.min.js"></script>`

The bugfix is working great!  I'm able to see the Score value in the popup now.  Thank you so much for pushing out the fix.

Like you said before, feel free to leave this issue open for the problem related to passing in a color function to the color field of the ROIs displayed at root level.
